doc/kernel/services/scheduling/index.rst

```diff
@@ -46,7 +46,9 @@ static priority, then the thread with the earlier deadline is considered
 to have the higher priority. Thus, when earliest-deadline-first scheduling is
 enabled, two threads are only considered to have the same priority when both
 their static priorities and deadlines are equal. The routine
-:c:func:`k_thread_deadline_set` is used to set a thread's deadline.
+:c:func:`k_thread_deadline_set` is used to set a thread's deadline. If a thread
+does not call :c:func:`k_thread_deadline_set`, its deadline is 0 which means this
+thread is executed before the others call :c:func:`k_thread_deadline_set`.
```

Comment:
> ... which means this thread is executed before the others call :c:func:`k_thread_deadline_set`.

It is not clear to me what this means. Do you mean that between two threads where one has set the deadline and one has not then the one that has not set the deadline will execute first? If so, that might not be true. My reading of the code in z_sched_prio_cmp() and the documentation of k_thread_deadline_set() suggests that 0 could be in the future if k_cycle_get_32() > 0x80000000.

Reply:
But in the [testcase](https://github.com/zephyrproject-rtos/zephyr/pull/69429/commits/9e4e5da1417c7de6e77481bca383cb9ac1dbffcd), It show the thread which do not call deadline_set run first.

Or I miss something?